hecuba_py/hecuba/hnumpy.py

```diff
@@ -107,14 +110,111 @@ def _store_meta(storage_args):
+        count_none = 0
+        for dim, coo in enumerate(coord):
+            if coo.stop is None and coo.start is None:
+                np_list.append([0, self.shape[dim]])
+                count_none = count_none + 1
+            elif coo.start is None:
+                np_list.append([0, coo.stop])
+            elif coo.stop is None:
+                np_list.append([coo.start, self.shape[dim]])
+            else:
+                np_list.append([coo.start, coo.stop])
+
+        if count_none == len(coord): return None
+        coordinates = np.array(np_list)
+        return coordinates
+
+    def slices_match_numpy_shape(self, sliced_coord):
+        if sliced_coord is None:
+            return True
+        elif len(self.shape) != len(sliced_coord):
+            return False
+        else:
+            for i, queried_slice in enumerate(sliced_coord):
+                return queried_slice[1] <= self.shape[i]
+
+    def get_coords_match_numpy_shape(self, coo):
+        new_coords = self.generate_coordinates(coo)
+        shape_slices = [slice(0, coord) for coord in self.shape]
+        formated_shape_coords = self.format_coords(shape_slices)
+        numpy_coords = self.generate_coordinates(formated_shape_coords)
+        coordinates = []
+        for coord in new_coords:
+            if coord in numpy_coords and coord not in coordinates:
+                coordinates.append(coord)
+        return coordinates
+
+    def __getitem__(self, sliced_coord):
+        log.info("RETRIEVING NUMPY")
+        if self._is_persistent:
+            # formats sliced coords
+            new_coords = self.format_coords(sliced_coord)
+
+            # checks if some coord in sliced_coords are inside the numpy
+            if not self.slices_match_numpy_shape(new_coords):  # some coordinates match
+                new_coords = self.get_coords_match_numpy_shape(new_coords)  # generates the coordinates
+                if not new_coords:
+                    self._hcache.load_numpy_slices([self.storage_id], [self.view(np.ndarray)],
+                                                   None)  # any coordinates generated match
+                    return super(StorageNumpy, self).__getitem__(sliced_coord)
+            else:  # coordinates match
+                new_coords = self.generate_coordinates(new_coords)
+
+            # coordinates is the union between the loaded coordiantes and the new ones
+            coordinates = list(set(it.chain.from_iterable((self._loaded_coordinates, new_coords))))
+
+            # checks if we already loaded the coordinates
+            if ((len(coordinates) != len(self._loaded_coordinates)) and not self._numpy_full_loaded) or (
+                    not self._numpy_full_loaded and not coordinates):
+                if not coordinates:
+                    self._numpy_full_loaded = True
+                    new_coords = None
+                self._hcache.load_numpy_slices([self.storage_id], [self.view(np.ndarray)], new_coords)
```

Comment:
`new_coords`is used to load the coordinates, but it does not check which coordinates were already loaded (and stored) in `_loaded_coordinates`
(or at least i could not see it)
Which makes me think that maybe coordinates are being loaded again?

Reply:
union between loaded_coord and new_coord:
    coordinates = list(set(it.chain.from_iterable((self._loaded_coordinates, new_coords))))
we will load coordinates if the union is different making the followin comparison:
    len(coordinates) != len(self._loaded_coordinates)

Reply:
The union goes to coordinates, which is a different variable than new_coords right?